katsdpcontroller/test/utils.py

```diff
@@ -369,3 +373,39 @@ async def replacement(*args, **kwargs):
     future = asyncio.Future()         # type: asyncio.Future[Any]
     mock.side_effect = replacement
     return future
+
+
+def timelimit(limit=5.0):
+    """Decorator to run tests with a time limit. It is designed to be used
+    with :class:`asynctest.TestCase`. It can be used as either a method or
+    a class decorator. It can be used as either ``@timelimit(limit)`` or
+    just ``@timelimit`` to use the default of 5 seconds.
+    """
+    if inspect.isfunction(limit) or inspect.isclass(limit):
+        # Used without parameters
+        return timelimit()(limit)
+
+    def decorator(arg):
+        if inspect.isclass(arg):
+            for key, value in arg.__dict__.items():
+                if (inspect.iscoroutinefunction(value) and key.startswith('test_')
+                        and not hasattr(arg, '_timelimit')):
+                    setattr(arg, key, decorator(value))
+            return arg
+        else:
+            @functools.wraps(arg)
+            async def wrapper(self, *args, **kwargs):
+                try:
+                    async with async_timeout.timeout(limit, loop=self.loop) as cm:
+                        await arg(self, *args, **kwargs)
+                except asyncio.TimeoutError:
+                    if not cm.expired:
+                        raise
+                    for task in asyncio.Task.all_tasks(loop=self.loop):
+                        if task.get_stack(limit=1):
+                            print()
```

Comment:
Not sure what this print() statement is doing?

Reply:
It puts a blank line between the stacks of the tasks so that one can tell where the boundaries are - otherwise it all just looks like one big stack trace.